kola/tests/kubeadm/kubeadm.go

```diff
@@ -210,9 +210,9 @@ func init() {
 				register.Register(&register.Test{
 					Name:    fmt.Sprintf("kubeadm.%s.%s%s.base", version, CNI, cgroupSuffix),
 					Distros: []string{"cl"},
-					// Network config problems in esx and qemu-unpriv
-					ExcludePlatforms: []string{"esx", "qemu-unpriv"},
 					// This should run on all clouds as a good end-to-end test
+					// Network config problems in qemu-unpriv
+					ExcludePlatforms: []string{"qemu-unpriv"},
```

Comment:
I thought we had other network problems, great if it was just the initrd

Reply:
This PR is old but now the feature has landed on stable, I'm preparing the merge. IIRC, this test was running fine. We just need to deal with LTS: I was thinking about adding a `Skip` on all tests impacted with the following condition: if "esx" and "lts" but it's not ideal.

Reply:
Ah, ok, we need our own DSL to evaluate test run conditions :D

Reply:
We could add an attribute that is a lambda function: `skipFunc: func(version, platform string)boolean{ … }`

Reply:
Ah yes, nice idea. We could even extend the signature to `func(version, channel, arch, platform string)`.

Reply:
With OVA from: https://stable.release.flatcar-linux.net/amd64-usr/3374.2.0/
```
kola run ... --platform=esx --channel=stable -esx-ova-path ./flatcar_production_vmware_ova.ova kubeadm.v1.24.1.calico.base
    --- PASS: kubeadm.v1.24.1.calico.base/node_readiness (38.29s)
    --- PASS: kubeadm.v1.24.1.calico.base/nginx_deployment (16.50s)
PASS, output in _kola_temp/esx-2022-11-17-1542-44142
```

Reply:
Glad to see that we can run kubeadm tests on one more platform!